.github/workflows/validate-and-test.yml

```
@@ -0,0 +1,73 @@
+name: Validate and test
+
+on:
+  pull_request:
+  push:
+    branches: [ master ]
+    tags-ignore: [ v* ]
+
+jobs:
+  formatting-check:
+    name: Check for missing formatting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up JDK ${{ matrix.JABBA_JDK }}
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.8
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.2
+
+      - name: Cache Build Target
+        uses: actions/cache@v2
+        with:
+          path: project/**/target
+          key: build-target-${{ hashFiles('**/*.sbt', 'project/build.properties', 'project/**/*.scala') }}
+
+      - name: Autoformat
+        run: sbt +headerCreateAll +scalariformFormat +test:scalariformFormat
+
+      - name: Check for missing formatting
+        run: git diff --exit-code --color || { echo "[error] Unformatted code found. Please run 'sbt +headerCreate +test:headerCreate +scalariformFormat +test:scalariformFormat' and commit the reformatted code."; false; }
+
+  compile-and-test:
+    name: Compile and test
```

```suggestion
    name: Test / Scala ${{ matrix.SCALA_VERSION }} / Java ${{ matrix.JABBA_JDK }}
```

Just to make it slightly easier to read checks on Github UI.

---

The Github UI automatically adds annotations for matrix parameters, so this would only add clutter (also not sure if it would work at the job level first of all).